iclib/bno055.py

```diff
@@ -0,0 +1,276 @@
+        except Exception as e:
+            _logger.error(f"Initialization failed: {str(e)}")
+            raise
+
+    def read_quaternion(self) -> list[int]:
+        data_w_lsb = self.read(Register.QUA_DATA_W_LSB, 1)
+        data_w_msb = self.read(Register.QUA_DATA_W_MSB, 1)
+        data_x_lsb = self.read(Register.QUA_DATA_X_LSB, 1)
+        data_x_msb = self.read(Register.QUA_DATA_X_MSB, 1)
+        data_z_lsb = self.read(Register.QUA_DATA_Z_LSB, 1)
+        data_z_msb = self.read(Register.QUA_DATA_Z_MSB, 1)
+        data_w_msb[0] = data_w_msb[0] * 256
+        data_x_msb[0] = data_x_msb[0] * 256
+        data_z_msb[0] = data_z_msb[0] * 256
+        return (
+            [data_w_lsb[0] + data_w_msb[0],
+             data_x_lsb[0] + data_w_msb[0],
+             data_z_lsb[0] + data_z_msb[0]]
+        )
+
+    def read_temperature(self) -> list[int]:
+        return self.read(Register.TEMP, 8)
+
+    def read_acceleration(self) -> list[float]:
+        accel_x_lsb = [
+            x / 100.0 for x in self.read(Register.ACC_DATA_X_LSB, 1)
+        ]
+        accel_y_lsb = [
+            x / 100.0 for x in self.read(Register.ACC_DATA_Y_LSB, 1)
+        ]
+        accel_z_lsb = [
+            x / 100.0 for x in self.read(Register.ACC_DATA_Z_LSB, 1)
+        ]
+        accel_x_msb = [
+            x / 100.0 for x in self.read(Register.ACC_DATA_X_MSB, 1)
+        ]
+        accel_y_msb = [
+            x / 100.0 for x in self.read(Register.ACC_DATA_Y_MSB, 1)
+        ]
+        accel_z_msb = [
+            x / 100.0 for x in self.read(Register.ACC_DATA_Z_MSB, 1)
+        ]
+        accel_x_msb[0] = accel_x_msb[0] * 256
+        accel_y_msb[0] = accel_y_msb[0] * 256
+        accel_z_msb[0] = accel_z_msb[0] * 256
+        return (
+            [accel_x_msb[0] + accel_x_lsb[0],
+             accel_y_msb[0] + accel_y_lsb[0],
+             accel_z_msb[0] + accel_z_lsb[0]]
+        )
+
+    def read_gyro(self) -> list[float]:
+        gyro_x_lsb = [
+            x / 16.0 for x in self.read(Register.GYR_DATA_X_LSB, 1)
+        ]
+        gyro_y_lsb = [
+            x / 16.0 for x in self.read(Register.GYR_DATA_Y_LSB, 1)
+        ]
+        gyro_z_lsb = [
+            x / 16.0 for x in self.read(Register.GYR_DATA_Z_LSB, 1)
+        ]
+        gyro_x_msb = [
+            x / 16.0 for x in self.read(Register.GYR_DATA_X_MSB, 1)
+        ]
+        gyro_y_msb = [
+            x / 16.0 for x in self.read(Register.GYR_DATA_Y_MSB, 1)
+        ]
+        gyro_z_msb = [
+            x / 16.0 for x in self.read(Register.GYR_DATA_Z_MSB, 1)
+        ]
+
+        gyro_x_msb[0] = gyro_x_msb[0] * 256
+        gyro_y_msb[0] = gyro_y_msb[0] * 256
+        gyro_z_msb[0] = gyro_z_msb[0] * 256
+        return (
+            [gyro_x_msb[0] + gyro_x_lsb[0],
+             gyro_y_msb[0] + gyro_y_lsb[0],
+             gyro_z_msb[0] + gyro_z_lsb[0]]
+        )
+
+    @property
+    def read_all_data(self) -> set[object]:
```

Comment:
Prefer `Any` over `object`. Also, in this case, should not be a set, preferably list of sub-return types. Even more preferably, create a separate dataclass with attributes; `acceleration`, `quaternion`, etc.